src/model/test/AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed_GTest.cpp

```diff
@@ -10,5 +10,338 @@
+  // Set
+  EXPECT_TRUE(hp.setSpeed4SupplyAirFlowRateDuringCoolingOperation(3.2));
+  ASSERT_TRUE(hp.speed4SupplyAirFlowRateDuringCoolingOperation());
+  EXPECT_EQ(3.2, hp.speed4SupplyAirFlowRateDuringCoolingOperation().get());
+  // Bad Value
+  EXPECT_FALSE(hp.setSpeed4SupplyAirFlowRateDuringCoolingOperation(-10.0));
+  ASSERT_TRUE(hp.speed4SupplyAirFlowRateDuringCoolingOperation());
+  EXPECT_EQ(3.2, hp.speed4SupplyAirFlowRateDuringCoolingOperation().get());
+  EXPECT_FALSE(hp.isSpeed4SupplyAirFlowRateDuringCoolingOperationAutosized());
+}
+
+TEST_F(ModelFixture, AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed_addToNode) {
+  Model m;
+  Schedule s = m.alwaysOnDiscreteSchedule();
+  FanConstantVolume supplyFan(m, s);
+
+  CoilCoolingDXMultiSpeed cc(m);
+  CoilHeatingElectricMultiStage hc(m);
+
+  CoilHeatingElectric suppHC(m);
+
+  AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed testObject(m, supplyFan, hc, cc, suppHC);
+
+  AirLoopHVAC airLoop(m);
+
+  Node supplyOutletNode = airLoop.supplyOutletNode();
+  EXPECT_EQ(2, airLoop.supplyComponents().size());
+  EXPECT_TRUE(testObject.addToNode(supplyOutletNode));
+  EXPECT_EQ(3, airLoop.supplyComponents().size());
+
+  EXPECT_EQ(5, airLoop.demandComponents().size());
+  Node inletNode = airLoop.zoneSplitter().lastOutletModelObject()->cast<Node>();
+  EXPECT_FALSE(testObject.addToNode(inletNode));
+  EXPECT_EQ(5, airLoop.demandComponents().size());
+
+  PlantLoop plantLoop(m);
+  supplyOutletNode = plantLoop.supplyOutletNode();
+  EXPECT_EQ(5, plantLoop.supplyComponents().size());
+  EXPECT_FALSE(testObject.addToNode(supplyOutletNode));
+  EXPECT_EQ(5, plantLoop.supplyComponents().size());
+
+  Node demandOutletNode = plantLoop.demandOutletNode();
+  EXPECT_EQ(5, plantLoop.demandComponents().size());
+  EXPECT_FALSE(testObject.addToNode(demandOutletNode));
+  EXPECT_EQ(5, plantLoop.demandComponents().size());
+
+  auto testObjectClone = testObject.clone(m).cast<AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed>();
+  // Clone should reset the inlet/outlet nodes
+  ASSERT_FALSE(testObjectClone.inletModelObject());
+  ASSERT_FALSE(testObjectClone.outletModelObject());
+
+  supplyOutletNode = airLoop.supplyOutletNode();
+  EXPECT_TRUE(testObjectClone.addToNode(supplyOutletNode));
+  EXPECT_EQ(5, airLoop.supplyComponents().size());
+}
+
+TEST_F(ModelFixture, AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed_Ctor_WrongChildType) {
+  Model m;
+  Schedule s = m.alwaysOnDiscreteSchedule();
+  FanConstantVolume supplyFan(m, s);
+
+  // Wrong cooling type, I expect a graceful failure, not a segfault
+  CoilCoolingDXSingleSpeed cc(m);
+  CoilHeatingElectricMultiStage hc(m);
+  CoilHeatingElectric suppHC(m);
+
+  // The segfault is caught here, but if graceful, the remove() was called!
+  EXPECT_THROW(AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed(m, supplyFan, hc, cc, suppHC), openstudio::Exception);
+  EXPECT_EQ(0, m.getConcreteModelObjects<AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed>().size());
+}
```

Comment:
Before fix, this doesn't segfault because there is a LOG_AND_THROW and not an OS_ASSERT, but the remove() is never called though.

```
[ RUN      ] ModelFixture.AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed_Ctor_WrongChildType
/media/DataExt4/Software/Others/OpenStudio2/src/model/test/AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed_GTest.cpp:346: Failure
Expected equality of these values:
  0
  m.getConcreteModelObjects<AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed>().size()
    Which is: 1
```